src/renderer/model/spotlight-layers.js

```diff
@@ -15,8 +17,25 @@ export default register => {
     : () => true
 
   const layerItem = (layerId, name, checked) => {
+
+    const exportLayer = (event) => {
+      event.stopPropagation()
+      electron.remote.dialog.showSaveDialog({ defaultPath: `layer-${layerId}.json` })
+        .then(result => {
+          if (result.canceled) return
+          try {
+            fs.writeFileSync(result.filePath, JSON.stringify(store.layer(layerId)))
+          } catch (error) {
+            console.error(error)
+          }
+        })
+        .catch(error => {
+          console.error(error)
+        })
+    }
+
     const [label, ...tags] = name.split(':')
-    const props = { label, tags: ['Layer', ...tags], checked }
+    const props = { label, tags: ['Layer', ...tags], checked, onSecondaryActionClicked: exportLayer }
```

Comment:
Propagating callback for secondary action would not be needed, if item directly sends an 'save/export event' to COMMAND_EXPORT_LAYER (either through IPC  or event bus, i.e. evented). I think this would decrease coupling by taking spotlight-layers out of the equation.

Reply:
Not sure if I get you right. LayerListItem is a pure React component and I don't want to pollute it with evented. Since export layer is a button we need to know what to do on click event. The only way I know about is to pass a handler function to the component. Do you have other suggestions?

Reply:
Hmm, what's 'pure' about the layer item? Its button's visual suggests there is some secondary action, namely export, which can be invoked. So its semantics is pretty much hard-wired already. 
Sure, we could supply one primary and an array of secondary actions (button icons along with side-effect functions), but then it would be a mere list item and no longer a __layer__ list item. NOTE: Spotlight layers already takes care about the primary action.

From another perspective: Is spotlight-layers more or less pure than layer item? Imagine the parent child hierarchy is deeper. I'd rather take the shortest (most direct route) to get things done. Bubbling up an event for 3 or more levels to eventually go down on another branch to render an effect does not sound that appealing to me.

Reply:
I called LayerListItem `pure` because it handles the visuals and wires interactions, but it does not care about business logic. You pass in props, but it does not handle its own state. This does not mean that is generic.
So in my opinion spotlight-layers is `less pure` (but maybe `pure` is the wrong terminus).
I don't see an event bubbling up 3 ore more levels, so why are we discussing it here?

However, I want to get things done. If you prefer firing the export command from LayerListItem I 'll implement it this way.

Reply:
In this case, where we have just one hop (item -> list) I see no real harm doing it your way. Also, this pattern is usually taught in React tuts. But in my opinion it should only be used with tightly coupled components (like it's the case here).

It's just that changing spotlight-layers was not necessary to implement this feature (given I/O is handled elsewhere).

Reply:
I changed the implementation and now LayerListItem calls COMMAND_EXPORT_LAYER itself. Please take a look.

Reply:
> In this case, where we have just one hop (item -> list) I see no real harm doing it your way. Also, this pattern is usually taught in React tuts. But in my opinion it should only be used with tightly coupled components (like it's the case here).
> 

I think this concept is used in React tutorials, among other things, because it increases the 'testability' of components (using unit tests).
If an end-to-end test approach is chosen this aspect might not be that important.

Reply:
Good point!
I have no experience unit testing React components.
I'm going to check/research the expected value of doing this.

Testability in this case could be improved by providing self-contained (secondary) actions, as mentioned above.